Automattic/wp-calypso
# WooCommerce: Add state for product listing

This PR adds a fetchProducts action, and related reducers and selectors for fetching a list of products.

Quick note on PR size: The actual logic/meat of this PR is only about ~180 lines. The rest is tests/test related code, and documentation.

This is patterned off of #14831 so the pagination system will work similarly for products and orders.  cc @ryelle as this also implements a way to get the total pages from the WooCommerce API.

To Test:
* `npm test` and make sure all tests pass (`npm run test-client client/extensions/woocommerce/state/sites/products` to run just products tests)
